caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/ChandraResolver.java

```diff
@@ -0,0 +1,77 @@
+/*
+ ************************************************************************
+ ****  C A N A D I A N   A S T R O N O M Y   D A T A   C E N T R E  *****
+ *
+ * (c) 2017.                            (c) 2017.
+ * National Research Council            Conseil national de recherches
+ * Ottawa, Canada, K1A 0R6              Ottawa, Canada, K1A 0R6
+ * All rights reserved                  Tous droits reserves
+ *
+ * NRC disclaims any warranties         Le CNRC denie toute garantie
+ * expressed, implied, or statu-        enoncee, implicite ou legale,
+ * tory, of any kind with respect       de quelque nature que se soit,
+ * to the software, including           concernant le logiciel, y com-
+ * without limitation any war-          pris sans restriction toute
+ * ranty of merchantability or          garantie de valeur marchande
+ * fitness for a particular pur-        ou de pertinence pour un usage
+ * pose.  NRC shall not be liable       particulier.  Le CNRC ne
+ * in any event for any damages,        pourra en aucun cas etre tenu
+ * whether direct or indirect,          responsable de tout dommage,
+ * special or general, consequen-       direct ou indirect, particul-
+ * tial or incidental, arising          ier ou general, accessoire ou
+ * from the use of the software.        fortuit, resultant de l'utili-
+ *                                      sation du logiciel.
+ *
+ ****  C A N A D I A N   A S T R O N O M Y   D A T A   C E N T R E  *****
+ ************************************************************************
+ */
+
+package ca.nrc.cadc.caom2.artifact.resolvers;
+
+import ca.nrc.cadc.caom2.artifact.resolvers.util.ResolverUtil;
+import ca.nrc.cadc.net.StorageResolver;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import org.apache.log4j.Logger;
+
+/**
+ * This class can convert a CHANDRA URI into a URL.
+ * NOTE: At time of first writing, format for CHANDRA URIs and URLs is not known
+ * so this class is a stub that will be updated as part of a future data engineering user story.
+ *
+ * @author jeevesh
+ */
+public class ChandraResolver implements StorageResolver {
+    public static final String SCHEME = "chandra";
+    private static final Logger log = Logger.getLogger(ChandraResolver.class);
+    private static final String BASE_ARTIFACT_URL = "http://";
+    private static final String CANNOT_GET_URL = "Can't generate URL from URI.";
+
+
+    public ChandraResolver() {
+    }
+
+    @Override
+    public String getScheme() {
+        return SCHEME;
+    }
+
+    @Override
+    public URL toURL(URI uri) {
+        ResolverUtil.validate(uri, SCHEME);
```

Comment:
I don't really see the point of implementing a stub for an essentially single-method class. Plus in this case it is going to throw a misleading RuntimeException. If we really need the stub, then toURL should just throw an UnsupportedOperationException.

Reply:
Could you elaborate on this, please?  The `toURL()` method is necessary as per the `StorageResolver`'s contract.  The `URL` instance needs to be created.  Are you suggesting this method gets simplified somehow?

Reply:
Since BASE_ARTIFACT_URL is not complete (ChandraResolver line 45) I assume this is a stub; that means that ResoverUtil line 68 is probably going to throw an exception that looks like a bug or bad input rather than the correct UnsupportedOperationException (correct for a stub).

If this impl is not supposed to be a stub, then BASE_ARTIFACT_URL needs to be fixed.